lib/node_modules/@stdlib/repl/lib/editor_actions.js

```diff
@@ -0,0 +1,580 @@
+	this._ttyWrite.call( this._rli, null, {
+		'sequence': '\u001F'
+	});
+});
+
+/**
+* Yank-pop the next string from the redo stack.
+*
+* @name redo
+* @memberof EditorActions.prototype
+* @type {Function}
+* @returns {void}
+*/
+setNonEnumerableReadOnly( EditorActions.prototype, 'redo', function redo() {
+	this._ttyWrite.call( this._rli, null, {
+		'sequence': '\u001E'
+	});
+});
+
+/**
+* Transposes the characters about cursor.
+*
+* @name transposeAboutCursor
+* @memberof EditorActions.prototype
+* @type {Function}
+* @returns {void}
+*/
+setNonEnumerableReadOnly( EditorActions.prototype, 'transposeAboutCursor', function transposeAboutCursor() {
+	var c = this._rli.line[ this._rli.cursor - 1 ];
+	this.deleteLeft();
+	this.moveRight();
+	this._rli.write( c );
+});
+
+/**
+* Changes the next word to uppercase.
+*
+* @name uppercaseNextWord
+* @memberof EditorActions.prototype
+* @type {Function}
+* @returns {void}
+*/
+setNonEnumerableReadOnly( EditorActions.prototype, 'uppercaseNextWord', function uppercaseNextWord() {
+	this._modifyNextWord( uppercase );
+});
+
+/**
+* Changes the next word to titlecase.
+*
+* @name capitalizeNextWord
+* @memberof EditorActions.prototype
+* @type {Function}
+* @returns {void}
+*/
+setNonEnumerableReadOnly( EditorActions.prototype, 'capitalizeNextWord', function capitalizeNextWord() {
+	this._modifyNextWord( capitalize );
+});
+
+/**
+* Changes the next word to lowercase.
+*
+* @name lowercaseNextWord
+* @memberof EditorActions.prototype
+* @type {Function}
+* @returns {void}
+*/
+setNonEnumerableReadOnly( EditorActions.prototype, 'lowercaseNextWord', function lowercaseNextWord() {
+	this._modifyNextWord( lowercase );
+});
+
+/**
+* Clears the entire REPL screen and scrollback history.
+*
+* @name clearScreen
+* @memberof EditorActions.prototype
+* @type {Function}
+* @returns {void}
+*/
+setNonEnumerableReadOnly( EditorActions.prototype, 'clearScreen', function clearScreen() {
+	this._ttyWrite.call( this._rli, null, {
+		'name': 'l',
+		'ctrl': true
+	});
+});
+
+/**
+* Callback which should be invoked **before** a "keypress" event is processed by a readline interface.
+*
+* @name beforeKeypress
+* @memberof EditorActions.prototype
+* @type {Function}
+* @param {string} data - input data
+* @param {(Object|void)} key - key object
+* @returns {boolean} boolean indicating whether an editor action was triggered
+*/
+setNonEnumerableReadOnly( EditorActions.prototype, 'beforeKeypress', function beforeKeypress( data, key ) {
+	var actionKeys;
+	var inputKeys;
+	var action;
+	var i;
+
+	if ( !key ) {
+		return false;
+	}
+	inputKeys = parseKey( key );
+	for ( i = 0; i < EDITOR_ACTIONS.length; i++ ) {
+		actionKeys = this._keybindings[ EDITOR_ACTIONS[ i ] ];
+		if ( !actionKeys ) {
+			continue; // no keybindings configured for the action
+		}
+		action = this._actions[ EDITOR_ACTIONS[ i ] ];
+		if ( matchKeybindings( inputKeys, actionKeys ) ) {
+			action.call( this );
```

Comment:
Not clear to me that this indirection and the maintenance of a lookup table is necessary. Why not

```suggestion
		if ( matchKeybindings( inputKeys, actionKeys ) ) {
			this[ EDITOR_ACTIONS[ i ] ]();
```

...given that the editor action name matches exactly to the method name.

Reply:
Yeah.. It's just that, relying on the name of the function felt kinda hacky..? Clearly mapping action-names to their corresponding functions felt more clear and robust to me. WDYT?😬

Reply:
Okay, what about a middle ground in which, instead of a table mapping `action -> method`, you can map `action -> method_name`. E.g., `moveRight` -> `moveRight`. This way you can have a variable in the parent scope for the table. E.g.,

```
var ACTIONS = {
	'moveRight': 'moveRight',
	...
};
```

Then in the keypress listener, you can do

```
this[ ACTIONS[ EDITOR_ACTIONS[ i ] ]();
```

Creating a separate effectively static table for every instance seems wasteful, and having to use `#.call( this )` seems unnecessary.

Reply:
The `action->method_name` table looks funny now😅. Let's just go with your previous solution of relying on function names..